I can't define a Numeric instance for a newtype of Double (Internal modules need to be exposed)

I'm stuck here (internal module, so I can't reach the type and tell it to just use Double for C calls, etc.):

```
src/HordeAd/Core/HasDual.hs:353:1: error:
    • Could not deduce (hmatrix-0.20.2:Internal.Numeric.CTrans
                          (Forward a))
        arising from the superclasses of an instance declaration
      from the context: (HM.Product (Forward a), HM.Container Vector a)
        bound by the instance declaration
        at src/HordeAd/Core/HasDual.hs:(353,1)-(354,43)
    • In the instance declaration for ‘Numeric (Forward a)’
    |
353 | deriving instance (HM.Product (Forward a), HM.Container Vector a)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

BTW, deriving with GHC 9.2.2

```
deriving instance (HM.Element a, Ord a) => HM.Element (Forward a)

deriving instance (HM.Container Vector a, Ord a, Integral a, Fractional a)
                  => HM.Container Vector (Forward a)
```

produces the following warnings:

```
src/HordeAd/Core/HasDual.hs:346:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Ord (Forward a)
    • In the type signature for:
           hmatrix-0.20.2:Internal.Matrix.sortI :: Ord (Forward a) =>
                                                   Vector (Forward a) -> Vector Foreign.C.Types.CInt
      When typechecking the code for ‘hmatrix-0.20.2:Internal.Matrix.sortI’
        in a derived instance for ‘HM.Element (Forward a)’:
        To see the code I am typechecking, use -ddump-deriv
      In the instance declaration for ‘HM.Element (Forward a)’
    |
346 | deriving instance (HM.Element a, Ord a) => HM.Element (Forward a)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/HordeAd/Core/HasDual.hs:346:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Ord (Forward a)
    • In the type signature for:
           hmatrix-0.20.2:Internal.Matrix.sortV :: Ord (Forward a) =>
                                                   Vector (Forward a) -> Vector (Forward a)
      When typechecking the code for ‘hmatrix-0.20.2:Internal.Matrix.sortV’
        in a derived instance for ‘HM.Element (Forward a)’:
        To see the code I am typechecking, use -ddump-deriv
      In the instance declaration for ‘HM.Element (Forward a)’
    |
346 | deriving instance (HM.Element a, Ord a) => HM.Element (Forward a)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/HordeAd/Core/HasDual.hs:346:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Ord (Forward a)
    • In the type signature for:
           hmatrix-0.20.2:Internal.Matrix.compareV :: Ord (Forward a) =>
                                                      Vector (Forward a)
                                                      -> Vector (Forward a)
                                                      -> Vector Foreign.C.Types.CInt
      When typechecking the code for ‘hmatrix-0.20.2:Internal.Matrix.compareV’
        in a derived instance for ‘HM.Element (Forward a)’:
        To see the code I am typechecking, use -ddump-deriv
      In the instance declaration for ‘HM.Element (Forward a)’
    |
346 | deriving instance (HM.Element a, Ord a) => HM.Element (Forward a)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/HordeAd/Core/HasDual.hs:348:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Ord (Forward a)
    • In the type signature for:
           hmatrix-0.20.2:Internal.Numeric.step' :: Ord (Forward a) =>
                                                    Vector (Forward a) -> Vector (Forward a)
      When typechecking the code for ‘hmatrix-0.20.2:Internal.Numeric.step'’
        in a derived instance for ‘HM.Container Vector (Forward a)’:
        To see the code I am typechecking, use -ddump-deriv
      In the instance declaration for ‘HM.Container Vector (Forward a)’
    |
348 | deriving instance (HM.Container Vector a, Ord a, Integral a, Fractional a)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...

src/HordeAd/Core/HasDual.hs:348:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Ord (Forward a)
    • In the type signature for:
           hmatrix-0.20.2:Internal.Numeric.ccompare' :: Ord (Forward a) =>
                                                        Vector (Forward a)
                                                        -> Vector (Forward a) -> Vector HM.I
      When typechecking the code for ‘hmatrix-0.20.2:Internal.Numeric.ccompare'’
        in a derived instance for ‘HM.Container Vector (Forward a)’:
        To see the code I am typechecking, use -ddump-deriv
      In the instance declaration for ‘HM.Container Vector (Forward a)’
    |
348 | deriving instance (HM.Container Vector a, Ord a, Integral a, Fractional a)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...

src/HordeAd/Core/HasDual.hs:348:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Fractional (Forward a)
    • In the type signature for:
           hmatrix-0.20.2:Internal.Numeric.scaleRecip :: Fractional
                                                           (Forward a) =>
                                                         Forward a
                                                         -> Vector (Forward a) -> Vector (Forward a)
      When typechecking the code for ‘hmatrix-0.20.2:Internal.Numeric.scaleRecip’
        in a derived instance for ‘HM.Container Vector (Forward a)’:
        To see the code I am typechecking, use -ddump-deriv
      In the instance declaration for ‘HM.Container Vector (Forward a)’
    |
348 | deriving instance (HM.Container Vector a, Ord a, Integral a, Fractional a)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...

src/HordeAd/Core/HasDual.hs:348:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Fractional (Forward a)
    • In the type signature for:
           hmatrix-0.20.2:Internal.Numeric.divide :: Fractional (Forward a) =>
                                                     Vector (Forward a)
                                                     -> Vector (Forward a) -> Vector (Forward a)
      When typechecking the code for ‘hmatrix-0.20.2:Internal.Numeric.divide’
        in a derived instance for ‘HM.Container Vector (Forward a)’:
        To see the code I am typechecking, use -ddump-deriv
      In the instance declaration for ‘HM.Container Vector (Forward a)’
    |
348 | deriving instance (HM.Container Vector a, Ord a, Integral a, Fractional a)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...

src/HordeAd/Core/HasDual.hs:348:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Fractional (Forward a)
    • In the type signature for:
           hmatrix-0.20.2:Internal.Numeric.arctan2' :: Fractional
                                                         (Forward a) =>
                                                       Vector (Forward a)
                                                       -> Vector (Forward a) -> Vector (Forward a)
      When typechecking the code for ‘hmatrix-0.20.2:Internal.Numeric.arctan2'’
        in a derived instance for ‘HM.Container Vector (Forward a)’:
        To see the code I am typechecking, use -ddump-deriv
      In the instance declaration for ‘HM.Container Vector (Forward a)’
    |
348 | deriving instance (HM.Container Vector a, Ord a, Integral a, Fractional a)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...

src/HordeAd/Core/HasDual.hs:348:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Integral (Forward a)
    • In the type signature for:
           hmatrix-0.20.2:Internal.Numeric.cmod' :: Integral (Forward a) =>
                                                    Forward a
                                                    -> Vector (Forward a) -> Vector (Forward a)
      When typechecking the code for ‘hmatrix-0.20.2:Internal.Numeric.cmod'’
        in a derived instance for ‘HM.Container Vector (Forward a)’:
        To see the code I am typechecking, use -ddump-deriv
      In the instance declaration for ‘HM.Container Vector (Forward a)’
    |
348 | deriving instance (HM.Container Vector a, Ord a, Integral a, Fractional a)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

--- a/packages/base/hmatrix.cabal
+++ b/packages/base/hmatrix.cabal
@@ -64,8 +64,7 @@ library
                         Numeric.LinearAlgebra.Data
                         Numeric.LinearAlgebra.HMatrix
                         Numeric.LinearAlgebra.Static
-
-    other-modules:      Internal.Vector
+                        Internal.Vector
                         Internal.Devel
                         Internal.Vectorized
                         Internal.Matrix

--- a/packages/base/src/Numeric/Vector.hs
+++ b/packages/base/src/Numeric/Vector.hs
@@ -20,7 +20,7 @@
 --
 -----------------------------------------------------------------------------
 
-module Numeric.Vector () where
+module Numeric.Vector (adaptScalar) where
 
 import Internal.Vectorized
 import Internal.Vector

